subsys/tracing/sysview/sysview.c

```diff
@@ -21,8 +21,12 @@ uint32_t sysview_get_timestamp(void)
 uint32_t sysview_get_interrupt(void)
 {
 #ifdef CONFIG_CPU_CORTEX_M
+	/* As per Cortex-M3 Devices Generic User Guide, we need to subtract 16 from
+	   the VECTACTIVE field of the ICSR regiester in order to obtain the CMSIS
+	   IRQ number.
+	*/
```

Comment:
The comment above does not really describe why we need to do this. Please revise it to something like: "The vector index for the Cortex-M interrupts start at 16 because the vectors 0-15 are used by exceptions."

I am not convinced we should be doing this though because Cortex-M, from implementation perspective, does not make a very clear distinction between "exceptions" and "interrupts," and there is no guarantee that we will never place the `sys_trace_isr_enter` and `sys_trace_isr_exit` inside the "exception" handlers.

For instance, the SysTick timer interrupt (at least, what would be an "interrupt" on other architectures), is really an exception with the vector index of 15. Although we do not call `sys_trace_isr_enter` and `sys_trace_isr_exit` inside the SysTick ISR right now (arguably due to an oversight), there is no reason why we must not do that, as with other architectures.

On Cortex-M, the "exceptions" should more appropriately be called "internal interrupts" (internal from NVIC perspective) and the "interrupts" should be called "external interrupts" (triggered via external IRQ lines).

Reply:
Thanks for the comment. I agree with you that this change is not needed. In addition to the reason you mentioned, the SystemView GUI lists the number as "ISR Number" and not "IRQ Number". And ISR number is well described in Table 2.5 in [this page ](https://developer.arm.com/documentation/dui0552/a/the-cortex-m3-processor/programmers-model/core-registers?lang=en) of the Cortex-M3 user guide.
So as long as the user is aware of the difference, this change is not necessary. I will close this PR.